pkg/proxy/util/utils.go

```diff
@@ -311,3 +322,44 @@ func EnsureSysctl(sysctl utilsysctl.Interface, name string, newVal int) error {
 	}
 	return nil
 }
+
+// DialContext is a dial function matching the signature of net.Dialer.DialContext.
+type DialContext = func(context.Context, string, string) (net.Conn, error)
+
+// FilteredDialOptions configures how a DialContext is wrapped by NewFilteredDialContext.
+type FilteredDialOptions struct {
+	// DialHostIPDenylist restricts hosts from being dialed. If this is nil, IsProxyableHostname is used.
+	DialHostCIDRDenylist []*net.IPNet
+}
+
+// NewFilteredDialContext returns a DialContext function that only allows connections to
+// proxyable addresses, as defined by IsProxyableHostname, in addition to the denylist in
+// FilteredDialOptions. If the wrapped DialContext is nil, all DNS and TCP settings use
+// golang default values.
+func NewFilteredDialContext(wrapped DialContext, opts *FilteredDialOptions) DialContext {
+	if wrapped == nil {
+		wrapped = http.DefaultTransport.(*http.Transport).DialContext
+	}
+	return func(ctx context.Context, network, address string) (net.Conn, error) {
+		resp, err := net.DefaultResolver.LookupIPAddr(ctx, address)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(resp) == 0 {
+			return nil, ErrNoAddresses
+		}
+
+		for _, host := range resp {
+			if err := isProxyableIP(host.IP); err != nil{
+				return nil, err
+			}
+			if opts != nil && opts.DialHostCIDRDenylist != nil {
+				if err := IsAllowedHost(ctx, host.IP, opts.DialHostCIDRDenylist); err != nil {
+				return nil, err
```

Comment:
nit: indentation (`hack/update-gofmt.sh` should fix it?)

Reply:
w/rt --dial-host-cidr-denylist=none, your idea is to make "none" the default value for the flag, so that to make it opt-out the only change will be for the flag default?